test_runner/fixtures/zenith_fixtures.py

```diff
@@ -810,48 +804,120 @@ class ZenithCli:
+        res = self.raw_cli(['tenant', 'create', '--tenant-id', tenant_id.hex])
+        res.check_returncode()
         return tenant_id
 
     def list_tenants(self) -> 'subprocess.CompletedProcess[str]':
-        return self.raw_cli(['tenant', 'list'])
+        res = self.raw_cli(['tenant', 'list'])
+        res.check_returncode()
+        return res
+
+    def create_timeline(self,
+                        new_branch_name: str,
+                        tenant_id: Optional[uuid.UUID] = None) -> uuid.UUID:
+        cmd = [
+            'timeline',
+            'create',
+            '--branch-name',
+            new_branch_name,
+            '--tenant-id',
+            (tenant_id or self.env.initial_tenant).hex,
+        ]
+
+        res = self.raw_cli(cmd)
+        res.check_returncode()
+
+        create_timeline_id_extractor = re.compile(r"^Created timeline '(?P<timeline_id>[^']+)'",
+                                                  re.MULTILINE)
+        matches = create_timeline_id_extractor.search(res.stdout)
+
+        created_timeline_id = None
+        if matches is not None:
+            created_timeline_id = matches.group('timeline_id')
+
+        if created_timeline_id is None:
+            raise Exception('could not find timeline id after `zenith timeline create` invocation')
+        else:
+            return uuid.UUID(created_timeline_id)
 
     def create_branch(self,
-                      branch_name: str,
-                      starting_point: str,
-                      tenant_id: Optional[uuid.UUID] = None) -> 'subprocess.CompletedProcess[str]':
-        args = ['branch']
-        if tenant_id is not None:
-            args.extend(['--tenantid', tenant_id.hex])
-        args.extend([branch_name, starting_point])
+                      new_branch_name: str = DEFAULT_BRANCH_NAME,
+                      ancestor_branch_name: Optional[str] = None,
+                      tenant_id: Optional[uuid.UUID] = None,
+                      ancestor_start_lsn: Optional[str] = None) -> uuid.UUID:
+        cmd = [
+            'timeline',
+            'branch',
+            '--branch-name',
+            new_branch_name,
+            '--tenant-id',
+            (tenant_id or self.env.initial_tenant).hex,
+        ]
+        if ancestor_branch_name is not None:
+            cmd.extend(['--ancestor-branch-name', ancestor_branch_name])
+        if ancestor_start_lsn is not None:
+            cmd.extend(['--ancestor-start-lsn', ancestor_start_lsn])
 
-        return self.raw_cli(args)
+        res = self.raw_cli(cmd)
+        res.check_returncode()
 
-    def list_branches(self,
-                      tenant_id: Optional[uuid.UUID] = None) -> 'subprocess.CompletedProcess[str]':
-        args = ['branch']
-        if tenant_id is not None:
-            args.extend(['--tenantid', tenant_id.hex])
-        return self.raw_cli(args)
+        create_timeline_id_extractor = re.compile(r"^Created timeline '(?P<timeline_id>[^']+)'",
```

Comment:
precompiled regex can be a top level const, or we can even introduce some flag to cli for machine readable output. For cases like this it can be `--quiet` flag and cli changed to output only id without other messages when this flag is supplied. Maybe this is redundant complication, but if someone changes wording in the output this will fail

Reply:
I've went with the global vars for simplicity 🙂